tests/test_grammar.py

```diff
@@ -767,3 +769,43 @@ def create() -> DictConfig:
         # Other invalid characters should be detected at creation time.
         with raises(GrammarParseError):
             create()
+
+
+def test_grammar_cache_is_thread_safe() -> None:
```

Comment:
how long does this test take to run?

Reply:
0.15s on my laptop when I run only this test (including pytest's overhead)

It's actually slower if I remove the `sleep()` calls because of Python's terrible multi-threading capabilities.